ingress/elb/elb.go

```diff
@@ -0,0 +1,136 @@
+package elb
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	aws_elb "github.com/aws/aws-sdk-go/service/elb"
+	"github.com/sky-uk/feed/ingress/types"
+)
+
+const (
+	elbTag = "KubernetesClusterFrontend"
+)
+
+// New  creates a new frontend
+func New(region string, clusterName string) types.Frontend {
+	metadata := ec2metadata.New(session.New())
+	return &elb{
+		metadata:    metadata,
+		awsElb:      aws_elb.New(session.New(&aws.Config{Region: &region})),
+		clusterName: clusterName,
+		maxTagQuery: 20,
+	}
+}
+
+type elb struct {
+	awsElb      ELB
+	metadata    EC2Metadata
+	clusterName string
+	maxTagQuery int
+}
+
+// ELB interface to allow mocking of real calls to AWS as well as cutting down the methods from the real
+// interface to only the ones we use
+type ELB interface {
+	DescribeLoadBalancers(input *aws_elb.DescribeLoadBalancersInput) (*aws_elb.DescribeLoadBalancersOutput, error)
+	DescribeTags(input *aws_elb.DescribeTagsInput) (*aws_elb.DescribeTagsOutput, error)
+	RegisterInstancesWithLoadBalancer(input *aws_elb.RegisterInstancesWithLoadBalancerInput) (*aws_elb.RegisterInstancesWithLoadBalancerOutput, error)
+	DeregisterInstancesFromLoadBalancer(input *aws_elb.DeregisterInstancesFromLoadBalancerInput) (*aws_elb.DeregisterInstancesFromLoadBalancerOutput, error)
+}
+
+// EC2Metadata interface to allow mocking of the real calls to AWS
+type EC2Metadata interface {
+	Available() bool
+	Region() (string, error)
+	GetInstanceIdentityDocument() (ec2metadata.EC2InstanceIdentityDocument, error)
+}
+
+func (e *elb) Attach(frontend types.FrontendInput) (int, error) {
+	log.Info("Attaching to loadbalancer with %v", frontend)
+
+	id, err := e.metadata.GetInstanceIdentityDocument()
+	if err != nil {
+		return 0, fmt.Errorf("unable to query ec2 metadata service for InstanceId: %v", err)
+	}
+
+	instance := id.InstanceID
+	log.Info("Ingress controller running on instance: ", instance)
```

Comment:
"Attaching to ELBs from instance %s", instanceId? Since it's about the elb attaching.